src/index.js

```diff
@@ -367,14 +369,21 @@ export default class extends Component {
         newState.offset = { x: 0, y: 0 }
         newState.offset[dir] = offset[dir] + 1
         this.setState(newState, () => {
+          this.props.onIndexUpdate(this.state.index, this.state.total)
```

Comment:
You are right that the index is actually changed in the first callback. My line of thinking was that the internal callback and the external one should be called consistently with each other. I think it could cause confusion that the external callback in some cases would be called with the internal, but in some cases would be invoked before.